### PR TITLE
add support for macsec

### PR DESCRIPTION
**Pull request by qsn**
For now this is only the part for creating macsec devices and configuring device attributes over RTNL-link.
The rest of the attributes of macsec devices is configured over genl, and is not part of this PR.

macsec devices are part of kernel 4.6-rc1.

### CODE REVIEW

File: libnl-route-3.sym [hunk 1]
```diff
@@ -396,6 +396,33 @@ global:
 	rtnl_link_is_veth;
 	rtnl_link_is_vlan;
 	rtnl_link_is_vxlan;
+	rtnl_link_macsec_alloc;
```
Comment (thom311) [inline libnl-route-3.sym]: These parts must be below in the libnl_3_2_28 section.

File: lib/route/link/macsec.c [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+	struct macsec_info *info = link->l_info;
+
+	IS_MACSEC_LINK_ASSERT(link);
+
+	info->window = window;
+
+	return 0;
+}
+
+uint32_t rtnl_link_macsec_get_window(struct rtnl_link *link)
```
Comment (thom311) [inline lib/route/link/macsec.c]: I know, there are many getters that follow this pattern. But this gives no way to return an error.

I'd prefer something like
  int rtnl_link_macsec_get_window(struct rtnl_link *link, uint32_t *window);
Like in commit 207dc0352779af46121a83fb4758cdd3155d9001

Also, usually rtnl objects have a mask field so you know which attributes are present or which not. I know it's cumbersome to add, but could we add that?

**Comment by qsn** [comment]
Thanks for the comments, Thomas.
I will add what you suggested. The mask would also fix the clang build error.

**Comment by thom311** [comment]
part of this was merged as https://github.com/thom311/libnl/pull/99

**Comment by thom311** [comment]
for the record: the other patches are now on https://github.com/thom311/libnl/pull/100

**Pull request closed without merging**